apps/learner/src/routes/(app)/+layout.svelte

```diff
@@ -1,24 +1,53 @@
 <script lang="ts">
   import { BookHeart, Compass, Home } from '@lucide/svelte';
+  import { onMount } from 'svelte';
 
   import { page } from '$app/state';
 
   const { children } = $props();
 
+  let isWithinViewport = $state(false);
+
   const isHomePage = $derived(page.url.pathname === '/');
   const isLearningPage = $derived(page.url.pathname === '/learning');
   const isExplorePage = $derived(page.url.pathname === '/explore');
+
+  let target: HTMLElement | null;
+
+  onMount(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      isWithinViewport = entry.isIntersecting;
+    });
+
+    if (target) {
+      observer.observe(target);
+    }
+
+    return () => {
+      if (target) {
+        observer.unobserve(target);
+      }
+    };
+  });
 </script>
 
-<nav class="z-1000 sticky px-6 py-5">
+<header
+  class="fixed inset-x-0 top-0 z-50 mx-auto max-w-5xl bg-slate-100/90 px-4 py-3 backdrop-blur-sm"
```

Comment:
Is _mx-auto_ still needed if we have _inset-x-0_?

Reply:
Good catch! The `mx-auto max-w-5xl px-4 py-3` should be inside of `header` and wraps around the content.